File: Tests/test7/distinct_lines.pl
#!/usr/bin/perl -w

use strict;

my %distinct_line = ();
my $n = $ARGV[0];
my $lineCount = 0;

while (my $line = <STDIN>) {
	$lineCount++;
	$line = lc($line);
	$line =~ s/^\s+//g;
	$line =~ s/\s+/ /g;
	$distinct_line{$line} += 1;

	my $distinct_counter = keys %distinct_line;

	if ($distinct_counter == $n) {
		print "$n distinct lines seen after $lineCount lines read.\n";
		exit;
	}
}
print "End of input reached after $lineCount lines read - $n different lines not seen.\n";

File: Tests/test09/distinct_lines.py
#!/usr/bin/python

import sys, re

numDistinct = int(sys.argv[1])
lineCount = 0
uniqueCount = 0
#done = 0

uniqueLine = []


while True:
	line = sys.stdin.readline()
	if (line == ""):
		break
	lineCount += 1
	line = re.sub("\s+"," ",line)
	line = re.sub("^\s","",line)
	line = re.sub("\s$","",line)
	line = line.lower()
	if line not in uniqueLine:
		uniqueLine.append(line)
		uniqueCount += 1
	if (uniqueCount == numDistinct):
		print ("%d distinct lines seen after %d lines read"%(numDistinct,lineCount))
		break

if (uniqueCount < numDistinct):
	print ("End of input reached after %d lines read -  %d different lines not seen."%(lineCount, numDistinct))
